pkg/model/workflow.go

```diff
@@ -272,23 +272,20 @@ func (j *Job) Needs() []string {
 	return nil
 }
 
-// RunsOn list for Job
-func (j *Job) RunsOn() []string {
+// RunsOn string or list for Job
```

Comment:
I considered leaving this method alone and always returning a `[]string`, and then in `runsOnPlatformNames` potentially spreading an expression array into `platformNames`, but that struck me as being less correct, since we wouldn't be able to differentiate between these two scenarios:

```yaml
# this is fine
runs-on: ${{ fromJSON('["my-label"]') }}
# this is not fine and will not work on GitHub
runs-on: [${{ fromJSON('["my-label"]') }}]
```

If we're not concerned about that and would prefer a slightly simpler implementation, I can tweak it

Reply:
Thanks for the yaml tip and the note about array merging, I'll go ahead simplify my change 🙏 

> I cannot even get my critical bugfix for windows merged or even reviewed with any comments, while a panic fix was merged

Oooof 😞

Reply:
Done ... though as you pointed out, matrix with templated runs-on is problematic (though that's the case before this change too), I'll file a separate issue for that, as I think the fix there will be a little more involved